### PR TITLE
feat(deps-usage), introduce "bit deps usage" to find components that use the given dependency

### PR DESCRIPTION
**Pull request by davidfirst**

### CODE REVIEW

File: scopes/dependencies/dependency-resolver/dependencies/dependency-list.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export class DependencyList {
   }
 
   findByPkgNameOrCompId(id: string, version?: string): Dependency | undefined {
-    const found = this.dependencies.find((dep) => dep.id === id || dep.getPackageName?.() === id);
+    const found = this.dependencies.find(
+      (dep) => dep.id === id || dep.getPackageName?.() === id || dep.id.startsWith(`${id}@`)
```
Comment (zkochan) [inline scopes/dependencies/dependency-resolver/dependencies/dependency-list.ts:54]: why was this change needed?

Comment (davidfirst) [reply scopes/dependencies/dependency-resolver/dependencies/dependency-list.ts:55]: @zkochan , because for some reason, components (not packages) are saved with versions as part of their ids.

Comment (davidfirst) [reply scopes/dependencies/dependency-resolver/dependencies/dependency-list.ts:55]: so let's say the dep.id is `button@0.1.0` and I'm searching for dependency "button" (without a version) I won't be able to find it otherwise

**Review by GiladShoham — APPROVED** [review]
